command/agent/command.go

```diff
@@ -396,12 +416,16 @@ func (c *Command) Run(args []string) int {
 	// Enable log streaming
 	logGate.Flush()
 
+	// Start retry join process
+	errCh := make(chan struct{})
```

Comment:
Can we add the retryJoinErr channel to the Command struct so we don't have to pass the channel into the functions